### PR TITLE
Add generic make target workflow executor for cmo

### PR DESCRIPTION
**Pull request by arajkumar**
This would enable us to hook `make versions` target by reusing most of the workflow part. I will create a follow PR to add workflow to perform `make versions` for CMO, which would update [jsonnet/versions.yaml](https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/versions.yaml) periodically.

### CODE REVIEW

**Comment by sthaha** [comment]
@arajkumar should this be on hold until the `make jsonnet-versions` pr is merged?

**Comment by arajkumar** [comment]
@sthaha  Actually it should go first, this is a dependency of `make jsonnet-versions`. :)

**Review by sthaha — APPROVED** [review]

**Pull request merged**
